### DISCUSSION TITLE
Why "max_worker_memory" config value refers to RSS instead of VSZ?

### DISCUSSION THREAD

**Discussion by lezhnev74** [roadrunner-server/roadrunner#1041] [Q&A] [answered]
Hey! 

I am digging into the internals of this program. I noticed it can check the memory consumption of a worker process and reload it automatically to avoid memory leaks. However internally I see it checks for RSS (which does not include swapped memory) instead of VSZ (which includes all allocated memory). I bet you had a reason to do so, can you elobarate? 

Thanks! :pray:

**Comment by rustatian** [comment, marked as answer]
@lezhnev74 Hey. RSS memory is the actual memory used by the application. VSZ on the other hand is just a memory that in total can be accessed by the application, but that's doesn't mean, that the memory is actually occupied. VSZ is always higher than the RSS and doesn't show the actual memory consumption.

**Reply by rustatian** [reply]
RSS memory can't be freed until it's used by the `MADV*` system calls for example. For example, if you have a `map` type that is actually garbage collected and not used anymore. In the case of VSZ, you'll see that freed up memory, but I guess, that's not actually relevant information for the user. For sure, we can add that info to the `rr workers` command output if such a request will be sent.
